src/formatDate.js

```diff
@@ -8,7 +8,48 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYearIndex = getIndex(fromFormat);
+  const newYearIndex = getIndex(toFormat);
+  const oldYearLength = fromFormat[oldYearIndex].length;
+  const newYearLength = toFormat[newYearIndex].length;
+  const oldJoin = fromFormat[3];
+  const newJoin = toFormat[3];
+  const newDate = date.split(oldJoin);
+  const reorderedDate = [];
+  const oldYear = newDate[oldYearIndex];
+
+  newDate[oldYearIndex] = getYear(oldYear, oldYearLength, newYearLength);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (i === newYearIndex) {
+      reorderedDate[i] = newDate[oldYearIndex];
+    } else {
+      const index = fromFormat.indexOf(toFormat[i]);
```

Comment:
The variable name `index` is not descriptive. It would be better to name it something like `datePartIndex` since it's specifically the index of a date part in the format array.